### PR TITLE
Bump Kubernetes dependencies to v1.28.5

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This PR updates Kubernetes dependencies to v1.28.5.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
This PR adopted the following changes:
- https://github.com/kubernetes/kubernetes/pull/116648
- https://github.com/kubernetes/kubernetes/pull/119207

**Does this PR introduce a user-facing change?**:
```release-note
Kubernetes dependencies now have been bumped to v1.28.5.
```

### CODE REVIEW

File: pkg/search/proxy/store/store.go [hunk 1]
```diff
@@ -191,6 +191,11 @@ func (s *store) Count(string) (int64, error) {
 	return 0, fmt.Errorf("count is not supported in proxy store")
 }
 
+// RequestWatchProgress implements storage.Interface.
+func (s *store) RequestWatchProgress(context.Context) error {
+	return fmt.Errorf("not implemented")
+}
+
```
Comment (RainbowMango) [inline pkg/search/proxy/store/store.go:198]: @XiShanYongYe-Chang @ikaven1024 @xigang 
Can you help to confirm if it OK to leave this interface unimplemented?
This interface was introduced by https://github.com/kubernetes/kubernetes/pull/119207.

File: hack/tools/swagger/lib/render.go [hunk 2]
```diff
@@ -80,6 +83,8 @@ func RenderOpenAPISpec(cfg Config) (string, error) {
 	}
 
 	serverConfig := genericapiserver.NewRecommendedConfig(cfg.Codecs)
+	serverConfig.ClientConfig = &clientgorest.Config{}
+	_ = utilfeature.DefaultMutableFeatureGate.Set(fmt.Sprintf("%s=%v", features.APIPriorityAndFairness, false))
```
Comment (RainbowMango) [inline hack/tools/swagger/lib/render.go:87]: @chaosi-zju 
Can you help to confirm if it is OK? and help to confirm the generated `swagger.json`?

File: pkg/search/proxy/store/store.go (hunk 1 above)
Comment (XiShanYongYe-Chang) [reply pkg/search/proxy/store/store.go:198]: By examining the calling path of the code, confirm that the current return of error has no impact. 

Test in the environment, and verify that the search function is working properly.

Comment (RainbowMango) [reply pkg/search/proxy/store/store.go:198]: Thanks.

**Pull request merged**